src/electrum/server.rs

```diff
@@ -490,6 +501,25 @@ impl Connection {
         Ok(result)
     }
 
+    fn log_rpc_event(&self, entries: &Vec<(&str, Value)>) {
+        let mut log = json!({});
+
+        if let Some(log_map) = log.as_object_mut() {
```

Comment:
This should never fail, I would consider using `unwrap()`

Reply:
Or alternatively construct `log` as a HashMap first, then convert to a json. This also somewhat simplifies the code:

```
fn log_rpc_event(&self, entries: Vec<(&str, Value)>) {
    let mut log = entries.into_iter().collect::<HashMap<_, _>>();
    log.insert(
        "source".into(),
        json!({
            "ip": self.addr.ip().to_string(),
            "port": self.addr.port(),
        }),
    );
    info!("{}", json!(log));
    }
```

Note that this requires passing ownership over `entries` from the macro, which would also reduce copying.

Reply:
Or perhaps even pass a `json!` map from the caller side to begin with and only inject `source` into it. This should simplify the code the most, as the `json!` syntax is more compact compared to the `Vec` construction.

Reply:
Suggested change applied.

Reply:
What my latest comment suggested was to pass a `serde::Value` map, not an `HashMap`:

```rust
fn log_rpc_event(&self, mut log: Value) {
    log.as_object_mut()
        .expect("log entry to be an object")
        .insert(
            "source".into(),
            json!({
                "ip": self.addr.ip().to_string(),
                "port": self.addr.port(),
            }),
        );
    println!("ELECTRUM-RPC-LOGGER:{}", log);
}
```

Then we could use the `json!()` syntax to construct the log entry, like so:
```rust
conditionally_log_rpc_event!(self, json!({
    "event": "rpc response",
    "payload_size": line.as_bytes().len(),
    "method": method_info,
}));
```

Reply:
Corrected, now it looks much more compact, thanks for comment!